fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/workflow/CacheMode.java

```diff
@@ -0,0 +1,17 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.handlers.workflow;
+
+import java.util.Optional;
+
+/**
+ * Cache Mode represents a strategy for Druid caching.
+ */
+public interface CacheMode {
+    /**
+     * Returns true if a particular cache mode is set.
+     *
+     * @return true if a particular cache mode is set
+     */
+    Optional<Boolean> isSet();
+}
```

Comment:
This optional isn't exactly what @cdeszaq and I were talking about adding 'isSet' we were actually thinking of adding it to the FeatureFlag interface so you could determine whether or not it was true.  Given that this 'isSet' is always defaulted to false, it could simply return boolean rather than optional boolean.